app/scripts/store/dimFarming.directive.js

```diff
@@ -0,0 +1,44 @@
+(function() {
+  'use strict';
+
+  angular.module('dimApp').directive('dimFarming', Farming);
+
+  function Farming() {
+    return {
+      controller: FarmingCtrl,
+      controllerAs: 'vm',
+      bindToController: true,
+      scope: {},
+      template: [
+        '<div id="item-farming" ng-if="vm.service.active">',
+        '  <div class="engram-icon">',
```

Comment:
The only awkward thing here is the engram icon... which I like too much to get rid of ;-)